pkg/controller/account/credentials_rotator.go

```diff
@@ -78,6 +79,17 @@ func (r *ReconcileAccount) RotateCredentials(reqLogger logr.Logger, awsSetupClie
 		return err
 	}
 
+	// Get latest version of the account
+	accountObjectKey, err := client.ObjectKeyFromObject(account)
+	if err != nil {
+		reqLogger.Error(err, "Unable to get name and namespace of Acccount object")
+	}
+
+	err = r.Client.Get(context.TODO(), accountObjectKey, account)
+	if err != nil {
+		reqLogger.Error(err, "Unable to get updated Acccount object after status update")
```

Comment:
We are not updating the status anywhere above here?

Reply:
we are not